src/hotspot/share/utilities/byteswap.hpp

```diff
@@ -63,7 +64,7 @@ struct ByteswapFallbackImpl;
 template <typename T>
 struct ByteswapFallbackImpl<T, 2> {
   inline constexpr uint16_t operator()(uint16_t x) const {
-    return (((x & UINT16_C(0x00ff)) << 8) | ((x & UINT16_C(0xff00)) >> 8));
+    return checked_cast<uint16_t>(((x & UINT16_C(0x00ff)) << 8) | ((x & UINT16_C(0xff00)) >> 8));
```

Comment:
What is the type of the expression without the cast?

Reply:
The type of the expression is `int`.

https://en.cppreference.com/w/cpp/language/implicit_conversion

> In particular, [arithmetic operators](https://en.cppreference.com/w/cpp/language/operator_arithmetic) do not accept types smaller than int as arguments, and integral promotions are automatically applied after lvalue-to-rvalue conversion, if applicable.